### DISCUSSION TITLE
https://topofmind.dev/blog/2023/08/25/profile-guided-optimizations-in-go/

### DISCUSSION THREAD

**Discussion by giscus** [LandonTClipp/LandonTClipp.github.io#10] [Announcements]
# https://topofmind.dev/blog/2023/08/25/profile-guided-optimizations-in-go/

How to use PGO to improve the performance of your production applications

https://landontclipp.github.io/blog/2023/08/25/profile-guided-optimizations-in-go/

**Comment by AndrewWPhillips** [comment]
Nice article.  I didn't notice any mistakes except that there were a few things missing.  Eg, you don't mention PGO code layout optimizations (possibly more important than even inlining).  Code layout is a general term for optimizations done in different parts of the build process (block reordering and function reordering) which have large benefits especially with improving the use of instruction cache.  (See my blog: https://andrewwphillips.github.io/blog/pgo.html#code-layout )

**Reply by LandonTClipp** [reply]
That's a great point, thanks for the shout. I'll look into updating this article to mention those types of optimizations!
